File: API/models/macroGoal.js
// imports mongoose to use to create schema
import mongoose from 'mongoose'

const goalSchema = new mongoose.Schema({
    startDate: {
        type: Date,
        required: true
    },
    endDate: {
        type: Date,
        required: false
    },
    calories: {
        type: Number,
        required: false
    },
    carbonhydrates: {
        type: Number,
        required: false
    },
    proteins: {
        type: Number,
        required: false
    },
    fats: {
        type: Number,
        required: false
    },
    margin: {
        type:Number,
        required: false,
        default: 5
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User"
    }
})

// exports scheme as model to controllers
const MacroGoal = mongoose.model('MacroGoal', goalSchema);
export default MacroGoal;

File: App/MealMate/MealMate/Models/MacroGoal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MealMate.Models
{
    public class MacroGoal
    {
        [Key]
        public string _id { get; set; }

        [Required]
        public DateTime startDate { get; set; }

        public DateTime? endDate { get; set; }

        public double? calories { get; set; }

        public double? carbohydrates { get; set; }

        public double? proteins { get; set; }

        public double? fats { get; set; }


        public int margin { get; set; }


        public User user { get; set; } 

    }

    public class MacroGoalResponse()
    {
        public MacroGoal macroGoal { get; set; }
    }
}
